samza-core/src/main/java/org/apache/samza/execution/ExecutionPlanner.java

```diff
@@ -323,7 +305,93 @@ private static void validateIntermediateStreamPartitions(JobGraph jobGraph) {
     }
   }
 
+  /**
+   * Ensures that all streams in supplied {@link JoinedStreamsGroup} agree in partition count.
+   * This includes setting partition counts of any intermediate streams that do not have their
```

Comment:
We are not setting the count for any intermediate streams here. It is only those streams that feed into joins that already have a source with known partition count.
Can we reflect that in the comment?

Reply:
We _are_ doing both verification and assignment in this method. Please, refer to my response to [this comment](https://github.com/apache/samza/pull/637/files#r219718076) for more details.